src/components/AttachmentModal.js

```diff
@@ -42,28 +45,91 @@ const propTypes = {
+    },
 };
 
 class AttachmentModal extends Component {
     constructor(props) {
         super(props);
 
+        this.handleKeyPress = this.handleKeyPress.bind(this);
+        this.getNextAttachment = this.getNextAttachment.bind(this);
+        this.getAttachmentSource = this.getAttachmentSource.bind(this);
+
         this.state = {
-            isModalOpen: false,
-            file: null,
-            sourceURL: props.sourceURL,
+            title: 'Attachment',
         };
     }
 
+    componentDidUpdate(prevProps) {
+        const {visibleAttachment: {isModalOpen}} = this.props;
+        if (!prevProps.visibleAttachment.isModalOpen && isModalOpen) {
+            document.addEventListener('keydown', this.handleKeyPress, false);
+        } else {
+            document.removeEventListener('keydown', this.handleKeyPress, false);
+        }
+    }
+
+    // Using regex to get the src attribute from the html img tag
+    getAttachmentSource(htmlString) {
+        const imgRegx = /<img.*?data-expensify-source="(.*?)"[^>]+>/g;
+        const imageGroup = imgRegx.exec(htmlString);
+        return imageGroup[1];
+    }
+
+    // A function to get the next attachment based on the current attachment being show in the report.
+    getNextAttachment = (toRight) => {
+        const {sortedReportActions, visibleAttachment: {sourceURL}} = this.props;
+        const attachments = sortedReportActions.filter(sortedReportAction => sortedReportAction.action.isAttachment);
+        if (attachments.length <= 1) {
+            return;
+        }
+
+        // Finding the current attachment inside the report using the sourceURL
+        const currentAttachment = attachments.find((attachment) => {
+            const html = lodashGet(attachment, ['action', 'message', 0, 'html'], '');
+            return html.includes(sourceURL);
+        });
+        const actions = [...sortedReportActions].reverse();
+        const sequenceNumber = currentAttachment.action.sequenceNumber;
+        const actionsToSearch = toRight
+            ? actions.slice(sequenceNumber)
+            : actions.slice(0, sequenceNumber !== 0 ? sequenceNumber - 1 : 0).reverse();
+        let nextAttachment = actionsToSearch.find(actionResult => actionResult.action.isAttachment);
+
+        // If there is no next attachment in the left or right direction,
+        // then we need to loop to get the next attachment after the end of the array
+        if (!nextAttachment) {
+            const actionsToSearchFromBeginning = toRight ? actions : [...actions].reverse();
+            nextAttachment = actionsToSearchFromBeginning.find(actionResult => actionResult.action.isAttachment);
+        }
+        const html = lodashGet(nextAttachment, ['action', 'message', 0, 'html'], '');
+        const attachmentSource = this.getAttachmentSource(html);
+        setAttachmentModalData({
+            sourceURL: attachmentSource, isAttachment: true, isModalOpen: true, file: null,
+        });
```

Comment:
I think we could simplify this process by just creating an array of actions _once_ (until the actions update again) and pass them to the modal as a prop instead of all actions. Then we can just find the index of our action in that small array. Look left and right +1/-1 and we've found the right action if it's out of bounds then return the attachment at `0` or `attachmentArray.length - 1`.

Reply:
the next action may not contain an attachment.

Reply:
Right, I am suggesting we _only_ pass actions with attachments and not all actions.

Reply:
ok